src/Database/DBStructure.php

```diff
@@ -1557,12 +1557,17 @@ public static function definition() {
 		$database["process"] = [
 				"comment" => "Currently running system processes",
 				"fields" => [
-						"pid" => ["type" => "int unsigned", "not null" => "1", "primary" => "1", "comment" => ""],
+						"id" => ["type" => "int unsigned", "not null" => "1", "extra" => "auto_increment", "primary" => "1", "comment" => "Global Unique ID"],
+						"pid" => ["type" => "int unsigned", "comment" => "PID of the process"],
+						"parent" => ["type" => "int unsigned", "comment" => "ID of the parent process"],
+						"type" => ["type" => "varchar(5)", "not null" => "1", "comment" => "Type of the process"],
 						"command" => ["type" => "varbinary(32)", "not null" => "1", "default" => "", "comment" => ""],
 						"created" => ["type" => "datetime", "not null" => "1", "default" => NULL_DATE, "comment" => ""],
 						],
 				"indexes" => [
-						"PRIMARY" => ["pid"],
+						"PRIMARY" => ["id"],
```

Comment:
I don't think that changing the primary index does work with our database update routine.

Reply:
What's the best way to change it? Adding a migration method?

Reply:
Currently there is no way. But possibly we should add some way. I already wanted to add some stuff like this changing of primary fields and removing of fields.

Reply:
So it's better to wait for your changes, isn't it? I don't want to include these routines in this PR.
Maybe we can add another PR with exactly this changes and merge it into this PR afterwards?

Reply:
It's complicated, especially since you want to add some auto-increment field. I have to see what we can do.

Reply:
I guess I will add the possibility to run a "pre update" script. There you have to ensure that the data and table structure is prepared for the change. This update script has to add the "id" field (not as primary at this moment), then it has to fill the existing lines. (We mustn't drop any task in the worker queue) After this is done it has to change the primary hey.

I don't think that an automatic update routine could handle this.

Reply:
That's bad because it seems like this PR can never get automatically deployed :-(

Would it be better to use `pid` for `id` and add a `process_id` for `pid`?
This seems really weird so I won't prefer it.

Another idea:
- set Friendica in maintenance mode
- stop all worker/daemon
- set `workerqueue`.`process_id` to 0
- drop `process`
- recreate `process`

Reply:
You could do this in a pre update script. But I don't think that this is a problem for automatic deployment. Why should it?

Reply:
I thought so because you wrote "I don't think that an automatic update routine could handle this." So I thought therefor it is a blocker for this PR "forever".

Maybe I used the wrong words, sorry

Reply:
With "automatic update routine" I meant the build in updater in the dbstructure class. This one won't be able to handle it. But a pre-update script should be able. But of course this needs to be tested very well.

Reply:
Sorry, I thought the pre-update is the update routine. That was the missing link :-)

Reply:
The current update routine can only do a post-update, no pre-update.